ftw/linkchecker/command/broken_link.py

```diff
@@ -28,12 +28,23 @@ def __init__(self):
         self.response_time = 'Unknown response time'
         self.error_message = 'No error occurred'
         self.creator = 'Unknown creator'
+        self.source_state = ''
 
     def __iter__(self):
         for attr in self.table_attrs:
             value = getattr(self, attr, '')
             yield value if isinstance(value, basestring) else ''
 
+    @staticmethod
+    def get_workflow_state(obj):
+        wftool = api.portal.get_tool('portal_workflow')
+        if len(wftool.getChainFor(obj)) <= 0:
```

Comment:
I'd not return the parent's workflow state if an object does not have a workflow assigned. How could we tell if it really is the block's workflow state or the parent's workflow state?

So I would return the workflow state only if the length of the workflow chain == 1. Otherwise (0 or >1) I'd return an empty string.

Reply:
Most Links will have `ftw.simplelayout.TextBlock` as their source object. Therefore I think the parents workflow should be listed in the exported file. I agree that your approach would be more accurate, but probably not very helpful.
Please correct me if I'm wrong, but wouldn't then be the new column mostly empty?

Reply:
Ok then...